src/equations/compressible_euler_2d.jl

```diff
@@ -339,10 +347,11 @@ Should be used together with [`TreeMesh`](@ref).
                                               surface_flux_function,
                                               equations::CompressibleEulerEquations2D)
     # get the appropriate normal vector from the orientation
+    RealT = eltype(u_inner)
     if orientation == 1
-        normal_direction = SVector(1, 0)
+        normal_direction = SVector(one(RealT), zero(RealT))
```

Comment:
we switch to `one(RealT), zero(RealT)`. What's the reason for this?

Since we are already making the code more complicated to read and write, wouldn't it be benficial to keep it simple here and keep the integer `SVector`?

Reply:
Here `normal_direction` has to be restricted to the specific type for calling `norm` please find discussion in slack.

Reply:
OK, thanks for the explanation

Reply:
Is this covered by a test, i.e., if someone were to make this wrong again, would we see it in CI?